lyric_finder/src/lib.rs

```diff
@@ -136,6 +138,45 @@ impl Default for Client {
     }
 }
 
+fn improve_query(query: &str) -> String {
+    let is_dash = |c: char| c == '-' || c == '—' || c == '–';
+    let r_no_filler = |s: &str, idx: usize| {
+        let Some(s) = s.get(..idx) else { return idx };
+        s.char_indices()
+            .rev()
+            .find(|(_, c)| !(is_dash(*c) || c.is_whitespace()))
+            .map_or(idx, |(idx, c)| idx + c.len_utf8())
+    };
+
+    let mut query = query.to_owned();
+    // just cut xxxx Remaster from query
+    if let Some(remaster_start) = query.find("Remaster").or_else(|| query.find("remaster")) {
```

Comment:
can process the query by `query = query.lower_bound()` so you don't need to manually handle cases like this

Reply:
Query is now always converted to lowercase. That shouldn't matter for search results.